File: BMI/app/src/main/java/com/example/bmi/ShowDishActivity.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class ShowDishActivity extends AppCompatActivity {
    private TextView dishReceip;
    private String group;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_dish);

        Intent intent = getIntent();
        String group = intent.getStringExtra("GROUP");

        dishReceip = (TextView) findViewById(R.id.textDish);

        if (group.equals("underweight")) {
            dishReceip.setText(getString(R.string.underweight_text));
        }
        else if (group.equals("optimum")) {
            dishReceip.setText(getString(R.string.optimum_text));
        }
        else if (group.equals("overweight")) {
            dishReceip.setText(getString(R.string.overweight_text));
        }
        else {dishReceip.setText(getString(R.string.default_text));}
    }
}

File: BMI_kotlin/app/src/main/java/com/example/bmi_kotlin/ShowDishActivity.kt
package com.example.bmi_kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_show_dish.*

class ShowDishActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_dish)

        val bundle : Bundle = intent.extras!!
        val group: String = bundle.getString("GROUP").toString()

        if (group == "underweight") textDish.text = getString(R.string.underweight_text)
        else if (group == "optimum") textDish.setText(getString(R.string.optimum_text))
        else if (group == "overweight") textDish.text = getString(R.string.overweight_text)
        else textDish.text = getString(R.string.default_text)
    }
}
